clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java

```diff
@@ -843,6 +844,20 @@ public GravitinoMetalake build() {
     }
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (!(o instanceof GravitinoMetalake)) {
+      return false;
+    }
+
+    GravitinoMetalake that = (GravitinoMetalake) o;
+    return super.equals(that) && Objects.equal(restClient, that.restClient);
```

Comment:
I feel like that we don't need to compare the `restClient` object. If the same metalake is gotten from different clients, I think though `restClient` is different, the `GravitinoMetalake` itself should be the same.

Reply:
Hi @jerryshao ,

Thanks for your suggestions!
I will remove the `restClient` comparison.

Best Regards,
Edward